irohad/network/impl/channel_factory.hpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_CHANNEL_FACTORY_HPP
+#define IROHA_CHANNEL_FACTORY_HPP
+
+#include "network/impl/channel_provider.hpp"
+
+#include <memory>
+#include <set>
+#include <string>
+
+#include <grpc++/grpc++.h>
+
+#include "ametsuchi/peer_query.hpp"
+#include "interfaces/common_objects/types.hpp"
+#include "network/impl/grpc_channel_params.hpp"
+
+namespace shared_model {
+  namespace interface {
+    class Peer;
+  }
+}  // namespace shared_model
+
+namespace iroha {
+  namespace network {
+
+    /**
+     * Creates client params which enable sending and receiving messages of
+     * INT_MAX bytes size with retries...
+     */
+    std::unique_ptr<GrpcChannelParams> getDefaultChannelParams();
+
+    grpc::ChannelArguments makeChannelArguments(
```

Comment:
It is not clear that it is required to declare all these methods in the header. Maybe try to reduce it?

Reply:
these appeared here when I needed to reuse some code in tests.